### PR TITLE
[1.13.x] Batch backports

### PR DESCRIPTION
**Pull request by MarianMacik**
Batch backport to 1.13.x.

Reviewer list:
* https://github.com/kiegroup/kogito-runtimes/commit/a6353799612dd4cd51106912f2a96de976c41ea1 - @danielezonca 
* https://github.com/kiegroup/kogito-runtimes/commit/2b5b76c6ac75ad51e7d432bd3a19a4639d6295d3 - @winklerm 
* https://github.com/kiegroup/kogito-runtimes/commit/773bff7e9d9abb678677bb0da0f180936d28b0d0 - @winklerm (instead of jstastny)
* https://github.com/kiegroup/kogito-runtimes/commit/909ad60a49d4b296913d434343461143b947f40a - @radtriste 
* https://github.com/kiegroup/kogito-runtimes/commit/ff8262b39cd1b67eb8d510d408e9d985278e9f5a - @ricardozanini 
* https://github.com/kiegroup/kogito-runtimes/commit/75d7a4eb10598817b69a148f347d523357859ecd - @ricardozanini 
* https://github.com/kiegroup/kogito-runtimes/commit/a04f37f194fc07d50b5256c1ecfe74196596b54b - @tarilabs 

Thanks!

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by kie-ci1** [comment]
**(tests) - kogito-runtimes job** `#49` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 2267
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.13.x/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/49/testReport/org.kie.kogito.quarkus.dmn/CustomEndpointIT/testHelloEndpoint/">org.kie.kogito.quarkus.dmn.CustomEndpointIT.testHelloEndpoint</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>JSON path d.Hello doesn't match.<br/>Expected: is "Hello Luca"<br/>  Actual: Certidão Hello Luca<br/><br/>	at org.kie.kogito.quarkus.dmn.CustomEndpointIT.testHelloEndpoint(CustomEndpointIT.java:39)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.13.x/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/49/display/redirect

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#54` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 415
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.13.x/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/54/testReport/org.acme.travel/MessagingIT/Build_projects___testProcess/">Build projects / org.acme.travel.MessagingIT.testProcess</a></summary>
org.opentest4j.AssertionFailedError: expected: <0> but was: <3><br/>	at org.acme.travel.MessagingIT.testProcess(MessagingIT.java:109)<br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.13.x/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/54/testReport/org.acme.travel/MessagingIT/Build_projects___testProcess/">Build projects / org.acme.travel.MessagingIT.testProcess</a></summary>
org.opentest4j.AssertionFailedError: expected: <0> but was: <1><br/>	at org.acme.travel.MessagingIT.testProcess(MessagingIT.java:93)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.13.x/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/54/display/redirect

File: api/kogito-api/src/test/java/org/kie/kogito/process/workitem/PoliciesTest.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
     @Test
     void testPolicies() {
         assertEquals(0, Policies.of(null).length);
-        Policy<IdentityProvider>[] policies = Policies.of("pepe", Arrays.asList("master", "of", "the", "universe"));
+        Policy<IdentityProvider>[] policies = Policies.of("pepe", Arrays.asList("chief", "of", "the", "universe"));
```
Comment (ricardozanini) [inline api/kogito-api/src/test/java/org/kie/kogito/process/workitem/PoliciesTest.java:30]: 👍

**Review by ricardozanini — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#56` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 416
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.13.x/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/56/testReport/org.kie.kogito.examples.demo/OutboxIT/testSendProcessEvents/">org.kie.kogito.examples.demo.OutboxIT.testSendProcessEvents</a></summary>
org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in org.kie.kogito.examples.demo.OutboxIT was not fulfilled within 4 minutes  10 seconds.<br/>	at org.kie.kogito.examples.demo.OutboxIT.testSendProcessEvents(OutboxIT.java:84)<br/>Caused by: java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <200> but was <404>.<br/><br/>	at org.kie.kogito.examples.demo.OutboxIT.lambda$testSendProcessEvents$1(OutboxIT.java:92)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.13.x/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/56/display/redirect

**Review by tarilabs — APPROVED** [review]
thank you

**Comment by MarianMacik** [comment]
Jenkins rerun kogito-examples tests

**Review by winklerm — APPROVED** [review]

**Pull request merged**